apps/backend/api/controllers/StripeController.js

```diff
@@ -0,0 +1,367 @@
+   *
+   * GET /noo/stripe/checkout/cancel
+   */
+  checkoutCancel: function (req, res) {
+    // In a real application, you might want to:
+    // 1. Log the cancellation
+    // 2. Show a message to the user
+    // 3. Redirect back to the product page
+
+    return res.json({
+      success: false,
+      message: 'Checkout was cancelled'
+    })
+  },
+
+  /**
+   * Webhook endpoint for Stripe events
+   *
+   * Stripe sends webhook events for important account and payment updates.
+   * This endpoint verifies the webhook signature and processes events.
+   *
+   * POST /noo/stripe/webhook
+   */
+  webhook: async function (req, res) {
+    try {
+      // Get the webhook signature from headers
+      const signature = req.headers['stripe-signature']
+
+      if (!signature) {
+        console.error('Missing Stripe signature header')
+        return res.status(400).json({ error: 'Missing signature' })
+      }
+
+      // Verify webhook signature
+      let event
+      try {
+        event = stripe.webhooks.constructEvent(
+          req.body,
+          signature,
+          process.env.STRIPE_WEBHOOK_SECRET
+        )
+      } catch (err) {
+        console.error('Webhook signature verification failed:', err.message)
+        return res.status(400).json({ error: 'Invalid signature' })
+      }
+
+      if (process.env.NODE_ENV === 'development') {
+        console.log(`Processing webhook event: ${event.type}`)
+      }
+
+      // Handle different event types
+      switch (event.type) {
+        case 'account.updated':
+          await this.handleAccountUpdated(event)
+          break
+
+        case 'payment_intent.succeeded':
+          await this.handlePaymentIntentSucceeded(event)
+          break
+
+        case 'payment_intent.payment_failed':
+          await this.handlePaymentIntentFailed(event)
+          break
+
+        case 'checkout.session.completed':
+          await this.handleCheckoutSessionCompleted(event)
+          break
+
+        default:
+          if (process.env.NODE_ENV === 'development') {
+            console.log(`Unhandled event type: ${event.type}`)
+          }
+      }
+
+      // Return a 200 response to acknowledge receipt of the event
+      return res.json({ received: true })
+    } catch (error) {
+      console.error('Webhook error:', error)
+      return res.status(400).json({
+        error: 'Webhook processing failed',
+        message: error.message
+      })
+    }
+  },
+
+  /**
+   * Handle account.updated webhook events
+   * Updates group's Stripe account status when onboarding changes
+   */
+  handleAccountUpdated: async function (event) {
+    try {
+      const account = event.data.object
+      if (process.env.NODE_ENV === 'development') {
+        console.log(`Account updated: ${account.id}`)
+      }
+
+      // Find the group with this Stripe account ID
+      const group = await Group.where({ stripe_account_id: account.id }).fetch()
```

Comment:
can multiple groups use the same stripe account? probably. i wonder if there is another identifier for like a sub-account or something like that? or maybe they just give separate account_ids for each one.

Reply:
ooo right that is an interesting question. This is a webhook, so I will have to check if we can store a 'foreign key' of sorts from our db in their records for an account. I assume we can